types/oidcsettings.pp

```diff
@@ -1,113 +1,148 @@
+    Optional['SessionInactivityTimeout']                   => Integer,
+    Optional['SessionMaxDuration']                         => Integer,
+    Optional['SessionType']                                => Pattern[/^(server-cache(:persistent)?|client-cookie(:persistent|:store_id_token|:persistent:store_id_token)?)$/],
+    Optional['SessionCacheFallbackToCookie']               => Enum['On', 'Off'],
+    Optional['CacheType']                                  => Enum['shm', 'memcache', 'file', 'redis'],
+    Optional['CacheDir']                                   => String,
+    Optional['CacheEncrypt']                               => Enum['On', 'Off'],
+    Optional['CacheShmMax']                                => Integer,
+    Optional['CacheShmEntrySizeMax']                       => Integer,
+    Optional['CacheFileCleanInterval']                     => Integer,
+    Optional['MemCacheServers']                            => String,
+    Optional['MemCacheConnectionsHMax']                    => Integer,
+    Optional['MemCacheConnectionsMin']                     => Integer,
+    Optional['MemCacheConnectionsSMax']                    => Integer,
+    Optional['MemCacheConnectionsTTL']                     => Integer,
+    Optional['RedisCacheServer']                           => String,
+    Optional['RedisCachePassword']                         => String,
+    Optional['RedisCacheConnectTimeout']                   => Pattern[/^[0-9]+\s?[0-9]*$/],
+    Optional['RedisCacheDatabase']                         => Integer,
+    Optional['RedisCacheTimeout']                          => Integer,
+    Optional['RedisCacheUsername']                         => String,
+    Optional['DiscoverURL']                                => Variant[Stdlib::HTTPSUrl, Stdlib::HttpUrl],
+    Optional['HTMLErrorTemplate']                          => String,
+    Optional['DefaultURL']                                 => Variant[Stdlib::HTTPSUrl, Stdlib::HttpUrl],
+    Optional['PathScope']                                  => Pattern[/^\"?[A-Za-z0-9\-\._\s]+\"?$/],
+    Optional['PathAuthRequestParams']                      => Pattern[/^[A-Za-z0-9\-\._%]+=[A-Za-z0-9\-\._%]+(&[A-Za-z0-9\-\._%]+=[A-Za-z0-9\-\._%]+)*$/],
+    Optional['IDTokenIatSlack']                            => Integer,
+    Optional['ClaimPrefix']                                => String,
+    Optional['ClaimDelimiter']                             => Pattern[/^.$/],
+    Optional['RemoteUserClaim']                            => String,
+    Optional['PassIDTokenAs']                              => Pattern[/^((claims|payload|serialized)\s?)+$/],
+    Optional['PassUserInfoAs']                             => Pattern[/^((claims|json(:([A-Za-z0-9\-\._])+)?|(signed_)?jwt(:([A-Za-z0-9\-\._])+)?)\s?)+$/],
+    Optional['PassClaimsAs']                               => Enum['none', 'headers', 'environment', 'both'],
```

Comment:
You wanted to update this one (another situation where RegExp is not suitable for rigorous validation of input but can do a vaguely correct job):

```suggestion
    Optional['PassClaimsAs']                               => Pattern[/^(none|headers|environment|both)?\s?(latin1|base64url|none)?$/],
```

Reply:
The [doc says](https://github.com/OpenIDC/mod_auth_openidc/blob/93bdf9d5427abe60e8e213e1f17b0af843107d58/auth_openidc.conf#L886):

> OIDCPassClaimsAs [none|headers|environment|both] [latin1|base64url|none]

My proposal above is to allow skipping one or the other of the optional parameters:

| Sample value | Current enum | Proposed regexp |
|---|:---:|:---:|
| `none` | :heavy_check_mark: | :heavy_check_mark: |
| `latin1` | :x: | :heavy_check_mark: |
| `none latin1` | :x: | :heavy_check_mark: |